File: fontmaniautil/src/main/java/in/januprasad/fontmaniautil/Font.java
package in.januprasad.fontmaniautil;

public class Font {

    private FontType type;

    public Font withType(FontType type) {
        this.type = type;
        return this;
    }

    public FontType getType() {
        return type;
    }

    public Font(FontType type) {

        this.type = type;
    }

}

File: fontmaniautil/src/main/java/in/januprasad/fontmaniautil/Font.kt
package `in`.januprasad.fontmaniautil

class Font(type: FontType) {

    var type: FontType? = null
        private set
    private val INSTANCE: Font

    init {
        this.type = type
        INSTANCE = this
    }

    fun withType(type: FontType): Font {
        INSTANCE.type = type
        return INSTANCE
    }

    override fun toString(): String {
        return type.toString()
    }

}